### ISSUE TITLE
Wrong order of file loading results in NameError

### ISSUE THREAD

**Issue by katafrakt** [polishgeeks/polishgeeks-dev-tools#39]
The commands are now loaded using this method: https://github.com/polishgeeks/polishgeeks-dev-tools/blob/master/lib/polishgeeks-dev-tools.rb#L22

The problem is that some commands depend on others (subclass them) and on the fact that they are already loaded. Unfortunately, the way it is done now does not guarantee that. Basically you can't rely on order of files returned by `Dir` in any way.

In my case, the outcome of this is:

```
rake aborted!
NameError: uninitialized constant PolishGeeks::DevTools::Commands::Rubocop
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polish_geeks/dev_tools/commands/rubocop_rspec.rb:6:in `<module:Commands>'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polish_geeks/dev_tools/commands/rubocop_rspec.rb:3:in `<module:DevTools>'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polish_geeks/dev_tools/commands/rubocop_rspec.rb:2:in `<module:PolishGeeks>'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polish_geeks/dev_tools/commands/rubocop_rspec.rb:1:in `<top (required)>'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polishgeeks-dev-tools.rb:25:in `require'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polishgeeks-dev-tools.rb:25:in `block (2 levels) in <top (required)>'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polishgeeks-dev-tools.rb:25:in `each'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polishgeeks-dev-tools.rb:25:in `block in <top (required)>'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polishgeeks-dev-tools.rb:23:in `each'
/home/katafrakt/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/bundler/gems/polishgeeks-dev-tools-f747d5d6fbb4/lib/polishgeeks-dev-tools.rb:23:in `<top (required)>'
```

Sample result of `Dir` method:

```
[3] pry(main)> base_path = File.dirname(__FILE__) + "/lib/polish_geeks/dev_tools/commands/**/*.rb"
=> "./lib/polish_geeks/dev_tools/commands/**/*.rb"
[4] pry(main)> Dir[base_path]
=> ["./lib/polish_geeks/dev_tools/commands/simplecov.rb",
 "./lib/polish_geeks/dev_tools/commands/final_blank_line.rb",
 "./lib/polish_geeks/dev_tools/commands/rubocop_rspec.rb",
 "./lib/polish_geeks/dev_tools/commands/rubocop.rb",
 "./lib/polish_geeks/dev_tools/commands/empty_methods.rb",
 "./lib/polish_geeks/dev_tools/commands/brakeman.rb",
 "./lib/polish_geeks/dev_tools/commands/expires_in.rb",
 "./lib/polish_geeks/dev_tools/commands/rspec.rb",
 "./lib/polish_geeks/dev_tools/commands/bundler_audit.rb",
 "./lib/polish_geeks/dev_tools/commands/yard.rb",
 "./lib/polish_geeks/dev_tools/commands/yml_parser.rb",
 "./lib/polish_geeks/dev_tools/commands/required_files.rb",
 "./lib/polish_geeks/dev_tools/commands/rspec_files_structure.rb",
 "./lib/polish_geeks/dev_tools/commands/base.rb",
 "./lib/polish_geeks/dev_tools/commands/examples_comparator.rb",
 "./lib/polish_geeks/dev_tools/commands/tasks_files_names.rb",
 "./lib/polish_geeks/dev_tools/commands/empty_methods/file_parser.rb",
 "./lib/polish_geeks/dev_tools/commands/empty_methods/string_refinements.rb",
 "./lib/polish_geeks/dev_tools/commands/allowed_extensions.rb",
 "./lib/polish_geeks/dev_tools/commands/rspec_files_names.rb",
 "./lib/polish_geeks/dev_tools/commands/rubycritic.rb",
 "./lib/polish_geeks/dev_tools/commands/haml_lint.rb"]
```

As you see, `rubocop_spec.rb` is above `rubocop.rb`, but the latter **must** be loaded before it.

The solution would be either to load all the files manually in particular order or to explicitly specify dependencies on top of command files, for example with

```
require_relative "./rubocop"
```

**Comment by nijikon** [comment]
@katafrakt thanks for your input, I will fix this.

**Comment by nijikon** [comment]
Fixed in 1.3.0.
